<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<title>The Statistical Bioinformatics Lab @ The Ohio State University </title>

<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<link rel="stylesheet" type="text/css" href="styles/flex_textfont_sans.css" media="all" />
<!-- SANDSTONE AND TILE TEMPLATE (FLEXIBLE-WIDTH, SANS-SERIF TEXT FONT VERSION) PROVIDED BY STANFORD UNIVERSITY INFORMATION TECHNOLOGY SERVICES FOR NON-DEPARTMENTAL OR UNOFFICIAL USE BY STANFORD FACULTY, STAFF AND STUDENTS. DESIGNED IN 2009 BY DAVE REAM, IT SERVICES. FOR POLICIES AND GUIDELINES ON ACCESSIBILITY, COPYRIGHT, AND ON THE USE OF STANFORD'S NAME AND TRADEMARKS, INCLUDING THE UNIVERSITY SEAL, SEE http://soap.stanford.edu, http://www.stanford.edu/group/identity, and http://adminguide.stanford.edu --> 
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-50750315-1', 'uiowa.edu');
  ga('send', 'pageview');
</script>
</head>


<body>

<!--BEGIN TOP RED BAR -->
<div id="top_navbar">
  <!-- OPTIONAL TOP NAVIGATION OR TEXT AREA INSIDE RED BAR; BAR EXPANDS HEIGHT AUTOMATICALLY TO ACCOMMODATE CONTENT -->
   <ul>
    <li><a href="http://www.augroup.org/">AU LAB - THE STATISTICAL BIOINFORMATICS LAB @ The Ohio State University</a></li>
  </ul>
  <!-- END OF OPTIONAL TOP RED BAR CONTENT -->
</div>
<!-- END TOP RED BAR -->

<div id="content_container">
  <!-- PAGE TITLE AREA AND BOX -->
   <div id="page_title_area">
    <!-- STANDARD PAGE TITLE; NO DECORATIVE GRAPHIC -->
    <!-- <div id="page_title_leftcontent"> -->
    <div id="page_title_leftcontent">
      <h1>Au Lab</h1>
      <ul>
    </div>


  </div>


  <!-- PAGE TITLE LEFT CONTENT WITH OPTIONAL QUAD LAMP DECORATION; UN-COMMENT TO USE -->


  <!-- END PAGE TITLE AREA AND BOX -->
  <div id="content">


</p>


<h2>Openings</h2> </p>
<ul>The Au Lab is seeking candidates for two postdoctoral research scholar positions to work on sequencing analyses:</ul>
    <ul>
	<li><b>1) Postdoctoral Research Scholar in bioinformatics method development:</b></li>
		<ul> We are looking for a postdoctoral scholar to develop statistical and computational/algorithmic methods for high-throughput sequencing data analyses, such as transcriptome and epigenetics using PacBio and Oxford Nanopore Technologies. Background on mathematics, statistics, computer science or degree in a related field is desirable.</ul>
		<li><b>2) Postdoctoral Research Scholar in computational biology</b></li>
		<ul>We are looking for a postdoctoral scholar in biology background. Our lab performs world leading science in development process and cancer research. Our team focuses on making meaningful advances to understand development process and cancer, using cutting edge technologies (third generation sequencing technology) to create authentic models and gain novel biological insight in transcriptome and epigenetics level. A typical candidate would have skills in including but not limited to implementing basic in-house bioinformatics tools, collaborating with statistician and computational biologists to elucidate biological discovery and mining out biological insights. Background on biology, computational biology or bioinformatics is desirable.
</ul>
        		
	<p>Strong ability in statistical analysis or computer programming (python/perl, C/C++, R, Matlab) is desirable.</p>
     The postdoctoral research scholars will have the opportunity to collaborate with internationally recognized biomedical scientists at University of Iowa Carver College of Medicine, Stanford University, University of Oxford and Pacific Biosciences.

	<br>    
    </ul>

    <p>
    	<br>
    </p>


 <h2>Latest Publications</h2>
		
<div><img src="images/CR.jpg" width="50" style='float: left;'/>
 <p><b>Single-molecule long-read sequencing reveals the chromatin basis of gene expression.</b>
<br><b>Wang, Y., Wang, A., Liu, Z.</b>, Thurman, A., Powers, L.S., Zou, M., Hefel, A., Li, Y., Zabner, J., <b>Au, K.F.</b>
  	<br><i> Genome Res.</i> 2019.  [<a href="https://genome.cshlp.org/content/early/2019/06/14/gr.251116.119.full.pdf+html">Manuscript</a>]
</p>
</div>
<table> 
	<tr><td>Genome-wide chromatin accessibility and nucleosome occupancy profiles have been widely investigated, while the long-range dynamics remains poorly studied at the single-cell level. Here we present a new experimental approach MeSMLR-seq (methyltransferase treatment followed by single-molecule long-read sequencing) for long-range mapping of nucleosomes and chromatin accessibility at single DNA molecules, and thus achieve comprehensive-coverage characterization of the corresponding heterogeneity. MeSMLR-seq offers direct measurements of both nucleosome-occupied and nucleosome-evicted regions on a single DNA molecule, which is challenging for many existing methods. We applied MeSMLR-seq to haploid yeast, where single DNA molecules represent single cells, and thus we could investigate the combinatorics of many (up to 356) nucleosomes at long range in single cells. We illustrated the differential organization principles of nucleosomes surrounding the transcription start site for silently- and actively-transcribed genes, at the single-cell level and in the long-range scale... 
</td>
	<td><img src="images/Single-mol.png" width="300"></td></tr> 
</table> 
<br>

	<div><img src="images/GenomeBio.jpg" width="50" style='float: left;'/>
 <p><b>A comparative evaluation of hybrid error correction methods for error-prone long reads</b> 
	<br>Fu, S., <b>Wang, A., Au, K.F.</b>
  	<br><i> Genome Biology.</i> 2019.  [<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1605-z">Manuscript</a>]
</p>
</div>
<table> 
	<tr><td>Third-generation sequencing technologies have advanced the progress of the biological research by generating reads that are substantially longer than second-generation sequencing technologies. However, their notorious high error rate impedes straightforward data analysis and limits their application. A handful of error correction methods for these error-prone long reads have been developed to date. The output data quality is very important for downstream analysis, whereas computing resources could limit the utility of some computing-intense tools. There is a lack of standardized assessments for these long-read error-correction methods. </td>
	<td><img src="images/ComparativeEval1.png" width="300"></td></tr> 
</table> 	

<br>
<div><img src="images/Sciencemag.jpg" width="55" style='float: left;'/>
 <p><b>EC coupling structural protein junctophilin-2 encodes a stress-adaptive transcription regulator.</b>
<br>Guo, A., Wang, Y., Chen, B., <b>Wang, Y. </b>,Yuan, J., Zhang, L., Hall, D., Wu, J., Shi, Y., Zhu, Q., Chen, C., Thiel, W.H., 
	Zhan, X., Weiss, R.M, Zhan, F., Musselman, C., Pufall, M., Zhu, W.,<b> Au, K.F. </b>, Hong, J., Anderson, M.E., Grueter, C.E., Song, L.S.
  	<br><i>Science.</i> 2018.  [<a href="https://science.sciencemag.org/content/sci/362/6421/eaan3303.full.pdf">Manuscript</a>]
</p>
</div>
<table> 
	<tr><td>Junctophilin-2 (JP2) is a structural protein required for normal excitation-contraction (E-C) coupling. After cardiac stress, JP2 is cleaved by the calcium ion-dependent protease calpain, which disrupts the E-C coupling ultrastructural machinery and drives heart failure progression. We found that stress-induced proteolysis of JP2 liberates an N-terminal fragment (JP2NT) that translocates to the nucleus, binds to genomic DNA, and controls expression of a spectrum of genes in cardiomyocytes. Transgenic overexpression of JP2NT in mice modifies the transcriptional profile, resulting in attenuated pathological remodeling in response to cardiac stress. Conversely, loss of nuclear JP2NT function accelerates stress-induced development of hypertrophy and heart failure in mutant mice. These data reveal a self-protective mechanism in failing cardiomyocytes that transduce mechanical information (E-C uncoupling) into salutary transcriptional reprogramming in the stressed heart.
</td>
	<td><img src="images/EC-coupling.png" width="300"></td></tr> 
</table> 		
	
<br>
<div><img src="images/genesMDPI.jpg" width="55" style='float: left;'/>
 <p><b>A Statistical Method for Observing Personal Diploid Methylomes and Transcriptomes with Single-Molecule Real-Time Sequencing.</b>
<br>Suzuki, Y., <b> Wang, Y., Au, K.F., </b> Morishita, S. 
	 <br><b># Co-corresponding author</b>
<br><i> Genes.</i> 2018.  [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6162384/pdf/genes-09-00460.pdf">Manuscript</a>]
</p>
</div>
<table> 
	<tr><td>We address the problem of observing personal diploid methylomes, CpG methylome pairs of homologous chromosomes that are distinguishable with respect to phased heterozygous variants (PHVs), which is challenging due to scarcity of PHVs in personal genomes. Single molecule real-time (SMRT) sequencing is promising as it outputs long reads with CpG methylation information, but a serious concern is whether reliable PHVs are available in erroneous SMRT reads with an error rate of ∼15%. To overcome the issue, we propose a statistical model that reduces the error rate of phasing CpG site to 1%, thereby calling CpG hypomethylation in each haplotype with >90% precision and sensitivity. Using our statistical model, we examined GNAS complex locus known for a combination of maternally, paternally, or biallelically expressed isoforms, and observed allele-specific methylation pattern almost perfectly reflecting their respective allele-specific expression status, demonstrating the merit of elucidating comprehensive personal diploid methylomes and transcriptomes.
</td>
	<td><img src="images/StatMethod.png" width="220"></td></tr> 
</table> 	
	
<br>
<div><img src="images/heliyon.jpg" width="55" style='float: left;'/>
 <p><b>Single cell expression analysis of primate-specific retroviruses-derived HPAT lincRNAs in viable human blastocysts identifies embryonic cells co-expressing genetic markers of multiple lineages.</b>
<br> Glinsky, G., Durruthy-Durruthy, J., Wossidlo, M., Grow, E.J., <b>Weirather, J.L.</b>,<b> Au, K.F.</b>, Wysocka, J., Sebastiano, V.
<br><i> Heliyon.</i> 2018.  [<a href="https://www.sciencedirect.com/science/article/pii/S2405844017328153">Manuscript</a>]
</p>
</div>
<table> 
	<tr><td>Chromosome instability and aneuploidies occur very frequently in human embryos, impairing proper embryogenesis and leading to cell cycle arrest, loss of cell viability, and developmental failures in 50–80% of cleavage-stage embryos. This high frequency of cellular extinction events represents a significant experimental obstacle challenging analyses of individual cells isolated from human preimplantation embryos. We carried out single cell expression profiling of 241 individual cells recovered from 32 human embryos during the early and late stages of viable human blastocyst (VHB) differentiation. Classification of embryonic cells was performed solely based on expression patterns of human pluripotency-associated transcripts (HPAT), which represent a family of primate-specific transposable element-derived lincRNAs highly expressed in human embryonic stem cells and regulating nuclear reprogramming and pluripotency induction...		</td>
	<td><img src="images/Single-cell.jpg" width="320"></td></tr> 
</table> 	

<br>
<div><img src="images/BriefingsBioinformatics.png" width="55" style='float: left;'/>
 <p><b>Revealing tumor heterogeneity of breast cancer by utilizing the linkage between somatic and germline mutations.</b>
<br><b>Zou, M., Jin, R., Au, K.F.#</b>
<br><i> Briefings in Bioinformatics.</i> 2018. (in press) [<a href="https://www.sciencedirect.com/science/article/pii/S2405844017328153">Manuscript</a>]
</p>
</div>
<table> 
	<tr><td>The intra-tumor heterogeneity is associated with cancer progression and therapeutic resistance, such as in breast cancer. While the existing methods for studying tumor heterogeneity only analyze variant allele frequency (VAF), the genotype of variant is also informative for inferring subclones, which can be detected by long reads or paired-end reads. We developed GenoClone to integrate VAF with the genotype of variant innovatively, so it showed superior performance of inferring the number of subclones, estimating the fractions of subclones and identifying somatic single-nucleotide variants composition of subclones. When GenoClone was applied to 389 TCGA breast cancer samples, it revealed extensive intra-tumor heterogeneity. We further found that a few somatic mutations were relevant to the late stage of tumor evolution, including the ones at the oncogene PIK3CA and the tumor suppress gene TP53. Moreover, 52 subclones that were identified from 167 samples shared high similarity of somatic mutations, which were clustered into three groups with the sizes of 24, 14 and 14... 		
		</td>
	<td><img src="images/GenoClone.png" width="250"></td></tr> 
</table> 
	
<br>
<div><img src="images/BioinfoCover.png" width="50" style='float: left;'/>
 <p><b>IDP-denovo: <i>de novo</i> transcriptome assembly and isoform annotation by hybrid sequencing.</b>
<br><b>Fu, S.</b>, <b>Ma Y.</b>, Yao, H., Xu, Z., Chen, S., Song, J., <b>Au, K.F.</b>
<br><i>Bioinformatics.</i> Volume 34, Issue 13, 2018, Pages 2168–2176. [<a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/bty098/4904264">Manuscript</a>]
</p>
</div>
<table> 
	<tr><td>In the past years, the long read (LR) sequencing technologies, such as Pacific Biosciences and Oxford Nanopore Technologies, have been demonstrated to substantially improve the quality of genome assembly and transcriptome characterization. Compared to the high cost of genome assembly by LR sequencing, it is more affordable to generate LRs for transcriptome characterization. That is, when informative transcriptome LR data are available without a high-quality genome, a method for de novo transcriptome assembly and annotation is of high demand.		
		</td>
	<td><img src="images/IDP-denovo.png" width="300"></td></tr> 
</table> 	

<p>
    	<br>
    </p>


	<h2>Research Focus</h2> </p>
    <ul>
	<li><b>RNA-seq and Third Generation Sequencing</b></li>
		<ul> We are interested in methodology research of Third Generation Sequencing (TGS) (especially for PacBio and Oxford Nanopore sequencing). 
		Au lab is working on both hybrid sequencing (Second Generation Sequencing (SGS) + TGS) and TGS-alone methodology research.
		Our research interests include but not limited to alternative splicing, isoform construction, gene fusion and quantitative analysis.</ul>
		<li><b>Stem cell transcriptome analysis</b></li>
		<ul>Au lab is applying the hybrid sequencing method on ESC, iPSC and preimplantation embryo, to deeply study the transcriptome differences between stem cells. </ul>
        <li><b>Proteomics</b> </li>
		<ul>Protein identification and novel splice detections from tandem Mass Spec are our research interests. Au lab is developing statistical methods for Integration of Mass Spec and sequencing data, in order to solve difficult proteomics problems.</ul>
		
	In addition, we are interested in the methodology research of gene regulatory network and RNA editing. 
	<br>    
    </ul>

    <p>
    	<br>
    </p>

	<h2>Collaborators</h2>
	Currently, we have very close ongoing collobrations with: 
	<ul><a href="http://www.stanford.edu/group/wonglab/">Wing H. Wong's Lab</a> at Stanford<br>
	<a href="http://www.pacificbiosciences.com/">Pacific Biosciences (PacBio)</a><br>
	<a href="http://www.binatechnologies.com/">Bina Technologies</a><br>
	<a href="http://stemcellphd.stanford.edu/faculty/renee-reijo-pera.html">Renee Reijo Pera's Lab</a> at Stanford<br>
    <a href="http://www.houstonmethodist.org/wing-tak-jack-wong-phd">Jack H. Wong's Lab</a> at Houston Methodist Research Institute<br>
    <a href="http://stemcellphd.stanford.edu/faculty/vittorio-sebastiano.html">Vittorio Sebastiano's Lab</a> at Stanford<br>
    <a href="http://www.well.ox.ac.uk/ogc/about-ogc">David Buck and the Oxford Genomics Centre</a> at the Wellcome Trust Centre for Human Genetics<br>
<a href="http://omicslab.genetics.ac.cn/index_en.php/">Xiujie Wang's Lab</a> at Beijing Institute of Genomics, Chinese Academy of Sciences<br>


<a href="http://sourcedb.big.cas.cn/yw/pl/fs/BIG_GPM/200907/t20090724_2194398.html">Yungui Yang's Lab</a> at Institute for Genetics and Developmental Biology, Chinese Academy of Sciences<br>

<a href="https://zhaohua.urmc.rochester.edu/">Xin Zhiguo Li's Lab</a> at University of Rochester<br>



    </ul>

    <p>
    	<br>
    </p>

    <h2>News</h2>
    <div>


<h3><img src="images/dna.png"> 06-15-2018: Shuhua Fu, Anqi Wang and Andrew Thurman are going to make presentations in ISMB 2018, July 7th, Chicago. </h3>
    <b>Topics:</b>
    <br>IDP-denovo: <i>de novo</i> transcriptome assembly and isoform annotation by hybrid sequencing. <b>(Shuhua Fu)</b>
    <br>Theoretical analysis of graph-based and alignment-based hybrid error correction methods for error-prone long reads. <b>(Anqi Wang)</b>
    <br>Gene isoform abundance quantification with Third Generation transcriptome sequencing. <b>(Andrew Thurman)</b>


<h3><img src="images/dna.png"> 02-23-2018: IDP-denovo paper is published </h3>
    <b>IDP-denovo: <i>de novo</i> transcriptome assembly and isoform annotation by hybrid sequencing.</b>
    <br><b>Fu, S.</b>, <b>Ma Y.</b>, Yao, H., Xu, Z., Chen, S., Song, J., <b>Au, K.F.</b> 
    <br><i>Bioinformatics</i> 2018. [<a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/bty098/4904264">Manuscript</a>]


    <h3><img src="images/dna.png"> 02-07-2016: AlignQC paper is released </h3>
    <b>Comprehensive comparison of Pacific Biosciences and Oxford Nanopore Technologies and their applications to transcriptome analysis [version 1; referees: awaiting peer review].</b>
    <br><b>Weirather, J.L.*</b>, de Cesare, M.*, <b>Wang, Y.*</b>, Piazza, P., Vittorio, S., Wang, X.J., Buck, D., <b>Au, K.F.</b>
    <br><i>F1000Research</i> 2017, 6:100 [<a href="http://dx.doi.org/10.12688/f1000research.10571.1">Manuscript</a>]
    <br><b>* These authors contributed equally to this work</b>
    <br>You can also check out the <a href="AlignQC/">Latest AlignQC Software</a> here if you are interested in using it to assess long read alignments.

    </div>
    <div>

    <h3><img src="images/dna.png"> 11-28-2016: IDP-ASE paper is released </h3>
    <b>IDP-ASE: haplotyping and quantifying allele-specific expression at the gene and gene isoform level by hybrid sequencing.</b><br>
    Deonovic, B., Wang, Y., Weirather, J., Wang, X.J., Au, K.F.
    <br><i>Nucleic Acids Research.</i> 2016. [<a href="http://nar.oxfordjournals.org/content/early/2016/11/28/nar.gkw1076">Manuscript</a>]

    <h3><img src="images/dna.png"> 07-08-2016: <a href="AlignQC/">AlignQC</a>, a tool for analyzing alignment files and assessing errors in alignments, is released.</h3>
    </div>
    
   
   
    
   
	<ul>
	
    </ul>
</p>
 
  </div>
  
<div id="sidebar">
    <h2>Site Map</h2>
    <ul>
      <li><a href="http://www.augroup.org">Home Page</a></li>
      <li><a href="http://www.augroup.org/Publications/Publications">Publications</a></li>
      <li><a href="http://www.augroup.org/labmembers/labmembers">Lab Members</a></li>
      <li><a href="http://www.augroup.org/utilities/utilities.html">Utilities</a></li>
      <li><a href="http://www.augroup.org/contact/contact">Contact</a></li>
    </ul>

    <h2>Software</h2>
    <ul>

    <li><a href="http://www.augroup.org/IDP/IDP">IDP</a> - Isoform Detection and Prediction
      <br><a href="http://www.augroup.org/IDP/IDP_download_current.html"><img src="images/download.png">IDP 0.1.8</a> (<a href="http://www.augroup.org/IDP/IDP-download-current.html#0.1">notes</a>)</li>

    <li><a href="http://www.augroup.org/IDP-ASE/IDP-ASE ">IDP-ASE</a> - IDP for Allele-Specific Expression
      <br><a href="http://www.augroup.org/IDP-ASE/IDP-ASE_download_current.html"><img src="images/download.png">IDP-ASE 20161017</a> (<a href="http://www.augroup.org/IDP-ASE/IDP-ASE-download-current.html#0.1">notes</a>)</li>
          
    <li><a href="http://www.augroup.org/IDP-fusion/IDP-fusion">IDP-fusion</a> - Fusion Isoform Detection and Prediction
      <br><a href="http://www.augroup.org/IDP-fusion/IDP-fusion/"><img src="images/download.png">IDP-fusion 1.1.1 </a> (<a href="http://www.augroup.org/IDP-fusion/IDP-fusion-download-current.html#0.1">notes</a>)</li>
      
 
    <li><a href="http://www.augroup.org/LSC/LSC">LSC</a> - Rapid error correction tool for PacBio data 
      <br><a href="http://www.augroup.org/LSC/LSC_download.html"><img src="images/download.png">LSC 2.0</a> (<a href="http://www.augroup.org/LSC/LSC_download.html#2.0">notes</a>)</li>
    
      <li><a href="http://www.stanford.edu/group/wonglab/SpliceMap/">SpliceMap</a> - De novo splice junction discovery and alignment tool 
    <br><a href="http://www.stanford.edu/group/wonglab/SpliceMap/download.html"><img src="images/download.png"> SpliceMap 3.3.5.2</a> (<a href="http://www.stanford.edu/group/wonglab/SpliceMap/download.html#3.3.5.2">notes</a>)</li>

      <li><a href="http://www.augroup.org/AlignQC/AlignQC_download_current.html">AlignQC</a> - Alignment quality control reporting tool 
    <br><a href="http://www.augroup.org/AlignQC/AlignQC_download_current.html"><img src="images/download.png"> AlignQC 1.2</a> (<a href="http://www.augroup.org/AlignQC/AlignQC_download_current.html#notes">notes</a>)</li>
    

<li><a href="http://www.augroup.org/IDP-denovo.20160829/IDP-denovo_download_current.html">IDP-denovo</a> - <i>de novo</i> transcriptome assembly and annotation tool 
    <br><a href="http://www.augroup.org/IDP-denovo.20160829/IDP-denovo_download_current.html"><img src="images/download.png"> IDP-denovo 0.2</a> (<a href="http://www.augroup.org/IDP-denovo.20160829/IDP-denovo_download_current.html#notes">notes</a>)</li>


<li><a href="http://www.augroup.org/GenoClone/GenoClone/GenoClone">GenoClone</a> - Reveal tumor heterogeneity
    <br><a href="http://www.augroup.org/GenoClone/GenoClone/GenoClone_download_current.html"><img src="images/download.png"> GenoCone 0.1</a> (<a href="http://www.augroup.org/GenoClone/GenoClone/GenoClone_download_current.html#notes">notes</a>)</li>

    </ul>
  
        <h2>Useful links</h2>
    <ul>
      <li><a href="http://www.augroup.org">Au Lab</a> - Au Lab homepage </li>
    </ul>
    <ul>
      <li><a href="http://www.stanford.edu/group/wonglab/SpliceMap/">SpliceMap</a> - Novel splice junction discovery</li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/LSC/LSC">LSC</a> - Error correction of PacBio long reads </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/IDP/IDP">IDP</a> - Isoform Detection and Prediction  </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/IDP-ASE/IDP-ASE">IDP-ASE</a> - Determine haplotypes and quantify allele specific expression  </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/IDP-fusion/IDP-fusion">IDP-Fusion</a> - Isoform Detection and Prediction across fusion genes  </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/IDP/hESC.html">hESC transcriptome</a> - hESC transcriptome by IDP </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/AlignQC/alignqc">AlignQC</a> - Check an alignment, assess error patterns, and make reports </li>
    </ul>
   

<ul>
      <li><a href="http://www.augroup.org/IDP-denovo.20160829/idp-denovo">IDP-denovo</a> - <i>de novo</i> transcriptome assembly and annotation</li>
    </ul>


<ul>
      <li><a href="http://www.augroup.org/GenoClone/GenoClone/GenoClone">GenoClone</a> - Reveal tumor heterogeneity</li>
    </ul>


 <ul>
      <li><a href="https://github.com/jason-weirather/Au-public">Au lab scripts</a> - Public repository of Au lab scripts and utilities </li>
    </ul>


  <br><ul><center><img src="images/seal.jpg" width="75%" hight="75%"><center></ul>
</div><div id="footer">
    <ul>
      <li>&copy; <a href=" http://www.augroup.org">Kin Fai Au</a>. All Rights Reserved.</li>
      <li><a href="https://www.osu.edu/">The Ohio State University</a>, OH 43210.</li>
    </ul>
</div>
  
  

<!-- end #content -->
</body>


<script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-23879262-1']);
  _gaq.push(['_trackPageview']);
  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script>


</html>
