<<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">

<head>
  <title>hESC transcriptome by IDP</title>
  
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<link rel="stylesheet" type="text/css" href="styles/flex_textfont_sans.css" media="all" />
<!-- SANDSTONE AND TILE TEMPLATE (FLEXIBLE-WIDTH, SANS-SERIF TEXT FONT VERSION) PROVIDED BY STANFORD UNIVERSITY INFORMATION TECHNOLOGY SERVICES FOR NON-DEPARTMENTAL OR UNOFFICIAL USE BY STANFORD FACULTY, STAFF AND STUDENTS. DESIGNED IN 2009 BY DAVE REAM, IT SERVICES. FOR POLICIES AND GUIDELINES ON ACCESSIBILITY, COPYRIGHT, AND ON THE USE OF STANFORD'S NAME AND TRADEMARKS, INCLUDING THE UNIVERSITY SEAL, SEE http://soap.stanford.edu, http://www.stanford.edu/group/identity, and http://adminguide.stanford.edu --> 
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-50750315-1', 'uiowa.edu');
  ga('send', 'pageview');

</script>
</head>


<body>

<!--BEGIN TOP RED BAR -->
<div id="top_navbar">
  <!-- OPTIONAL TOP NAVIGATION OR TEXT AREA INSIDE RED BAR; BAR EXPANDS HEIGHT AUTOMATICALLY TO ACCOMMODATE CONTENT -->
   <ul>
    <li>This hESC transcriptome iS identified by <a href="http://www.stanford.edu/group/wonglab/SpliceMap/">SpliceMap</a>-<a href="http://www.augroup.org/LSC/">LSC</a>-<a href="http://www.augroup.org/IDP/">IDP</a> pipeline</li>
  </ul>
  <!-- END OF OPTIONAL TOP RED BAR CONTENT -->
</div>
<!-- END TOP RED BAR -->


<div id="content_container">
  <!-- PAGE TITLE AREA AND BOX -->
    <div id="page_title_area">
    <!-- STANDARD PAGE TITLE; NO DECORATIVE GRAPHIC -->
    <!-- <div id="page_title_leftcontent"> -->
    <div id="page_title_leftcontent">
      <h1>hESC transcriptome</h1>
      <ul>
        <li>This Human Embryonic Stem Cell <br>(hESC, H1 cell line) <br>transcriptome is identified by <a href="http://www.stanford.edu/group/wonglab/SpliceMap/"><b>SpliceMap</b></a>-<a href="http://www.augroup.org/LSC/"><b>LSC</b></a>-<a href="http://www.augroup.org/IDP/"><b>IDP</b></a> pipeline<br>

      </ul>
    </div>
    
    <div id="page_title_rightcontent">
  <ul>


      <li>&nbsp;</li>

    <li><img src="images/IDP.png" width="300" >&nbsp;&nbsp;&nbsp;&nbsp;</li>
    
  </ul>
    </div>
    
    
  </div>


  <!-- PAGE TITLE LEFT CONTENT WITH OPTIONAL QUAD LAMP DECORATION; UN-COMMENT TO USE -->


  <!-- END PAGE TITLE AREA AND BOX -->
  <div id="content">
    <h2>hESC transcriptome - H1 cell line</h2>
    <p>This website contains the data and the results in each section of <a href="http://www.augroup.org/pubs/IDP.pdf">our PNAS paper</a>. 
	You can click the section title to find the updated data, figures, tables and other results. </p>
    <ul>
        <li><b><a href="hESC_Gene_isoform_detection_and_prediction.html">Gene isoform detection and prediction</a></b></li>
			<dd> 8,084 detections and 10,811 predictions of gene isoforms. Overall identification rate is 74%. </dd>		
        <li><b><a href="hESC_Novel_genes.html">Novel genes</a></b></li>
			<dd> More than 100 novel isoforms are identied in the novel gene loci, which are not reported by existing annotations (RefSeq, Ensembl, KnownGen and Gencode).</dd>		
        <li><b><a href="hESC_hESC-specific_novel_genes.html">hESC-specific novel genes</a></b></li>
			<dd> 23 novel genes (Human Pluripotent Assoiated Transcript, HPAT) have specificly high expression in hESC. </dd>		
        <li><b><a href="hESC_Novel_gene_isoforms.html">Novel gene isoforms</a></b></li>
			<dd> 2,103 novel isoforms are idenfied and are not reported bexisting annotations (RefSeq, Ensembl, KnownGen and Gencode).</dd>		
		<li><b><a href="hESC_Quantification_of_isoform_abundance.html">Quantification of isoform abundance</a></b></li>
			<dd> Gene isoform abundance estimation by using this identified transcriptome.</dd>		
		<li><b><a href="hESC_Isoforms_of_pluripotent_stem_cell_markers.html">Isoforms of pluripotent stem cell markers</a></b></li>
			<dd> The expressed isoforms (including novel ones) of the pluripotent stem cell markers, such as Oct4, Sall4, Sox2 ...</dd>		
		<li><b><a href="hESC_ncRNA_identification.html">ncRNA identification</a></b></li>
			<dd> 116 novel lncRNAs are identified and 68 have relatively high expressions in hESC.</dd>		
		<li><b><a href="hESC_Data.html">Data</a></b></li>
			<dd> Millions of PacBio long reads and 100 million of Illumina short reads.</dd>		
        <li><b><a href="hESC_howitworks.html">How it works? </a></b></li>
			<dd> The SpliceMap-LSC-IDP pipeline is applied to identify gene isoforms. </dd>		
			</ul>
	    <p> <font color="red"> We keep optimizing the results by running the updated LSC and IDP. More and more reliable results will be updated. If you want to subscribe our update news, please contact Kin Fai Au: </font> <img src="images/contact.png"></p>

    <h2>Latest publications</h2>

    <p>
    Durruthy-Durruthy J., Sebastiano V., Wossidlo M., Cepeda D., Cui J., Grow E.J., Davila J., Mall M., Wing W.H., Wysocka J., <b>Au, K.F.</b>, Pera, R.R.
     <br><b>A novel primate-specific noncoding RNA modulates human embryo- and pluripotent stem cell fate.</b><br>
     <i>Nature Genetics.</i> 2015. In press.
    </p>
<br>
    <p>
    Kin Fai Au, Vittorio Sebastiano, Pegah Tootoonchi Afshar, Jens Durruthy Durruthy, Lawrence Lee, Brian A. Williams, Honoratus Van Bakel, Eric Schadt, Renee A. Reijo Pera, Jason Underwood, Wing Hung Wong
<br>
    <b> Characterization of the human ESC transcriptome by hybrid sequencing </b> [<a href="http://www.augroup.org/pubs/IDP.pdf">preprint</a>] 
	<br><i>Proc. Natl. Acad. Sci. USA</i> 2013 110 (50) E4821-E4830 [<a href="http://www.augroup.org/pubs/IDP.pdf">preprint</a>] <br>
	<br>
    <i></i>  
  </p>

	<div id="SpliceMap-LSC-IDP pipeline">
    <h2>SpliceMap-LSC-IDP pipeline</h2>
    </div>
    <p>This hESC transcriptome is identified by <a href="http://www.stanford.edu/group/wonglab/SpliceMap/">SpliceMap</a>-<a href="http://www.augroup.org/LSC/">LSC</a>-<a href="http://www.augroup.org/IDP/">IDP</a> pipeline.</p>
	<p>SpliceMap takes short reads from the Second Generation Sequencing platforms, such as Illumina, to detect exon junctions.</p>
	<p>LSC makes use of the high-quality short reads to correct the long reads from PacBio platform. The output is the error-corrected long reads.</p>
	<p>IDP uses the junction detections and the alignment of error-corrected long reads to detect the relatively short isoforms at full-length and predict the very long isoforms by statistical modeling. </p>
	<p></p>
	<center><img src="images/SpliceMap-LSC-IDP_pipeline.png" width="100%" hight="100%" >&nbsp;&nbsp;&nbsp;&nbsp;</center> 


	
    <div id="news">
    <h2>Latest News</h2>
    </div>


    <div id="0.3">

	

	<p>
   <h3><img src="images/dna.png"> 11-26-2013: hESC transcriptome and IDP paper is released </h3>
 Kin Fai Au, Vittorio Sebastiano, Pegah Tootoonchi Afshar, Jens Durruthy Durruthy, Lawrence Lee, Brian A. Williams, Honoratus Van Bakel, Eric Schadt, Renee A. Reijo Pera, Jason Underwood, Wing Hung Wong
<br>
    <b> Characterization of the human ESC transcriptome by hybrid sequencing </b> 
    </p>
	
	
     </div>

    </p>

	
    <p>
      If you want to subscribe our update news, please contact Kin Fai Au: <img src="images/Picture3.png" width="20%" hight="20%">

.
   </p>
   
  </div>

<!-- include virtual="sidebar.inc" -->

  

<div id="sidebar">
    <h2>Site Map</h2>
    <ul>
      <li><a href="hESC.html">Home</a></li>
      <li><a href="hESC_Gene_isoform_detection_and_prediction.html">Gene isoform detection and prediction</a></li>
      <li><a href="hESC_hESC-specific_novel_genes.html">hESC-specific novel genes</a></li>
      <li><a href="hESC_Novel_gene_isoforms.html">Novel gene isoforms</a></li>
    <li><a href="hESC_Quantification_of_isoform_abundance.html">Quantification of isoform abundance</a></li>
    <li><a href="hESC_Isoforms_of_pluripotent_stem_cell_markers.html">Isoforms of pluripotent stem cell markers</a></li>
    <li><a href="hESC_ncRNA_identification.html">ncRNA identification</a></li>
    <li><a href="hESC_Data.html">Data</a></li>
    
    </ul>
    <h3>Help</h3>
    <ul>
    <li><a href="hESC_gpd_format.html">GPD format</a></li>
      <li><a href="hESC_howitworks.html">How it works?</a></li>
      <li><a href="http://www.augroup.org/IDP/hESC_long_read_file_guide.html">Generate long read files</a></li>
      <li><a href="http://www.augroup.org/utility_script.html">Utilities</a></li>
      <li><a href="hESC_faq.html">FAQ</a></li>
      <li><a href="hESC_contact.html">Contact</a></li>
    </ul>

        <h2>Useful links</h2>
       <ul>
      <li><a href="http://www.augroup.org">Au Lab</a> - Au Lab homepage </li>
    </ul>
    <ul>
      <li><a href="http://www.stanford.edu/group/wonglab/SpliceMap/">SpliceMap</a> - Novel splice junction discovery</li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/LSC/LSC">LSC</a> - Error correction of PacBio long reads </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/IDP/IDP">IDP</a> - Isoform Detection and Prediction  </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/IDP-ASE/IDP-ASE">IDP-ASE</a> - Determine haplotypes and quantify allele specific expression  </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/IDP-fusion/IDP-fusion">IDP-Fusion</a> - Isoform Detection and Prediction across fusion genes  </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/IDP/hESC.html">hESC transcriptome</a> - hESC transcriptome by IDP </li>
    </ul>
    <ul>
      <li><a href="https://github.com/jason-weirather/Au-public">Au lab scripts</a> - Public repository of Au lab scripts and utilities </li>
    </ul>


    <br><ul><center><img src="images/seal.jpg" width="75%" hight="75%"><center></ul>
<div id="footer">
    <ul>
      <li>&copy; <a href=" http://www.augroup.org">Kin Fai Au</a>. All Rights Reserved.</li>
      <li><a href="https://www.osu.edu/">The Ohio State University</a>, OH 43210.</li>
    </ul>
</div>
<!-- end #content -->
</body>


</html>
