<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">

<head>
  <title>IDP download</title>
  
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<link rel="stylesheet" type="text/css" href="styles/flex_textfont_sans.css" media="all" />
<!-- SANDSTONE AND TILE TEMPLATE (FLEXIBLE-WIDTH, SANS-SERIF TEXT FONT VERSION) PROVIDED BY STANFORD UNIVERSITY INFORMATION TECHNOLOGY SERVICES FOR NON-DEPARTMENTAL OR UNOFFICIAL USE BY STANFORD FACULTY, STAFF AND STUDENTS. DESIGNED IN 2009 BY DAVE REAM, IT SERVICES. FOR POLICIES AND GUIDELINES ON ACCESSIBILITY, COPYRIGHT, AND ON THE USE OF STANFORD'S NAME AND TRADEMARKS, INCLUDING THE UNIVERSITY SEAL, SEE http://soap.stanford.edu, http://www.stanford.edu/group/identity, and http://adminguide.stanford.edu --> 
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-50750315-1', 'uiowa.edu');
  ga('send', 'pageview');
</script>
</head>

<body>

<!--BEGIN TOP RED BAR -->
<div id="top_navbar">
  <!-- OPTIONAL TOP NAVIGATION OR TEXT AREA INSIDE RED BAR; BAR EXPANDS HEIGHT AUTOMATICALLY TO ACCOMMODATE CONTENT -->
   <ul>
    <li>IDP IS DEVELOPED BY <a href="http://www.augroup.org/">KIN FAI AU</a> with help from collaborators</li>
  </ul>
  <!-- END OF OPTIONAL TOP RED BAR CONTENT -->
</div>
<!-- END TOP RED BAR -->

<div id="content_container">
  <!-- PAGE TITLE AREA AND BOX -->

    <div id="page_title_area">
    <!-- STANDARD PAGE TITLE; NO DECORATIVE GRAPHIC -->
    <!-- <div id="page_title_leftcontent"> -->
    <div id="page_title_leftcontent">
      <h1>IDP</h1>
      <ul>
        <li>IDP is an gene <b>Isoform Detection and Prediction</b> tool <br> from Second Generation Sequencing and PacBio sequencing. <br>
        It offers very reliable gene isoform identification <br>
		with high sensitivity <br>

      </ul>
    </div>
    
    <div id="page_title_rightcontent">
	<ul>


	    <li>&nbsp;</li>

		<li><img src="images/IDP.png" width="300" >&nbsp;&nbsp;&nbsp;&nbsp;</li>
		
	</ul>
    </div>

<p class="page_title_area_lower">Latest News:  IDP 0.1.6 major update is released ... <a href="IDP.html#news">read more</a></p>
    
    
  </div>

  <!-- END PAGE TITLE AREA AND BOX -->
  <div id="content">
    <h2>Downloads</h2>
    <p>
    Copyright and installation instructions are contained within the package. We <b>strongly
    recommend</b> that you consult the <a href="IDP_tutorial.html">tutorial</a> before attempting to use IDP. 
    </p>
        
    <table border="1" cellpadding="1">
    <tr>
    <td>
    <a href="https://github.com/Dingjie-Wang/Manual-for-running-IDP-pipeline"><img src="images/download.png"><b>IDP 0.1.9</b></a>
    </td>
     <td>
    Please see the <a href="IDP_download_current.html#notes">release notes</a> for 
    more information
    </td>
    </tr>

    
    </table>
	
    <div id="notes">
    <h3><img src="images/dna.png">  <b>IDP 0.1.9</b> - Release Notes</h3>
    </div>
   <p>
   IDP integrates short reads (e.g. Illumina data) and long reads (e.g. PacBio data) to identify gene isoforms (transcripts) from transcriptome (see Figure above).<br>

  
    </p>
    <ul>
    <li> One input of IDP is the short-read RNA-seq results: junctions (bed file) AND alignments of short reads (sam file). 
	<br> Most RNA-seq tools, such as <a href="http://www.stanford.edu/group/wonglab/SpliceMap/">SpliceMap</a> and Tophat can output these two files.</li>
    <li> The other input is the long reads: raw sequences (FASTA file) OR alignment of long reads (PSL file by BLAT or <a href="IDP_gpd_format.html">GPD file</a>)
	<br> The error-corrected long reads from PacBio data is perferred. <a href="http://www.augroup.org/LSC/LSC/">LSC</a> is our default error-correction tool. </li>
	
	<li> The IDP output are the gene isoform identifications and quantification of genes and gene isoforms. <a href="hESC.html"> hESC transcriptome (H1 cell line)</a> is the first one identified by this methods. 
	For more details of this transcriptome, please see its homepage <a href="hESC.html"> <b>http://www.augroup.org/IDP/hESC.html</b> </a> and our paper <b> Characterization of the human ESC transcriptome by hybrid sequencing </b>[<a href="http://www.augroup.org/IDP/IDP.pdf">preprint</a>].
  </ul>
    <h3>Changes in version 0.1.9<br></h3>
<ul>
  <li> We added support for setting a hard cutoff for the fraction total gene expression to consider an isoform as a candidate (min_isoform_fraction).  This is mutually exclusive with FPR so please use one or the other. We also added support for a hard RPKM cutoff of isoform expression to be considered as an isoform candidated (min_isoform_rpkm).</li>
  <li> We added the utility IDP_merge_genepred.py to facilitate the comparison of multiple IDP runs.  Since each prediction can create uniquely named loci, this will merge them into a comparable format.</li>
</ul>

  </div>
  
  
<div id="sidebar">
    <h2>Site Map</h2>
    <ul>
      <li><a href="IDP.html">Home</a></li>
      
      <li><a href="IDP_download_current.html">Downloads</a></li>
      <li><a href="IDP_contact.html">Contact</a></li>
      <li><a href="IDP_requirements.html">Requirements</a></li>
      <li><a href="IDP_faq.html">FAQ</a></li>
  
    </ul>
    <h3>Help</h3>
    <ul>
      
     

      <li><a href="IDP_manual.html">Manual</a></li>
      <li><a href="IDP_tutorial.html">Tutorial</a></li>
      
    </ul>
    <h2>Download</h2>
    <h3>Current release</h3>
    <ul>
      <li><a href="IDP_download_current.html"><img src="images/download.png"> IDP 0.1.9</a> (<a href="IDP_download_current.html#notes">notes</a>)</li>
    </ul>

<h2>Useful links</h2>
    <ul>
      <li><a href="http://www.augroup.org">Au Lab</a> - Au Lab homepage </li>
    </ul>
    <ul>
      <li><a href="http://www.stanford.edu/group/wonglab/SpliceMap/">SpliceMap</a> - Novel splice junction discovery</li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/LSC/LSC">LSC</a> - Error correction of PacBio long reads </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/IDP/IDP">IDP</a> - Isoform Detection and Prediction  </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/IDP-ASE/IDP-ASE">IDP-ASE</a> - Determine haplotypes and quantify allele specific expression  </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/IDP-fusion/IDP-fusion">IDP-Fusion</a> - Isoform Detection and Prediction across fusion genes  </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/IDP/hESC.html">hESC transcriptome</a> - hESC transcriptome by IDP </li>
    </ul>
    <ul>
      <li><a href="http://www.augroup.org/AlignQC/alignqc">AlignQC</a> - Check an alignment, assess error patterns, and make reports </li>
    </ul>
   

<ul>
      <li><a href="http://www.augroup.org/IDP-denovo.20160829/idp-denovo">IDP-denovo</a> - <i>de novo</i> transcriptome assembly and annotation</li>
    </ul>


<ul>
      <li><a href="http://www.augroup.org/GenoClone/GenoClone/GenoClone">GenoClone</a> - Reveal tumor heterogeneity</li>
    </ul>

 <ul>
      <li><a href="https://github.com/jason-weirather/Au-public">Au lab scripts</a> - Public repository of Au lab scripts and utilities </li>
    </ul>


    <br><ul><center><img src="images/seal.jpg" width="75%" hight="75%"><center></ul>
  </div>


<div id="footer">
    <ul>
      <li>&copy; <a href=" http://www.augroup.org">Kin Fai Au</a>. All Rights Reserved.</li>
      <li><a href="https://www.osu.edu/">The Ohio State University</a>, OH 43210.</li>
    </ul>
</div>
  
  
<!-- end #content -->
</body>

</html>
